lib/system/supports-colors.js

```diff
@@ -49,12 +45,16 @@ module.exports = (function () {
     return true;
   }
 
-  if (process.env.TERM === 'dumb') {
+  if (/^screen|^xterm|^vt100|color|ansi|cygwin|linux/i.test(process.env.TERM)) {
+    return true;
+  }
+
+  if (process.stdout && !process.stdout.isTTY) {
     return false;
   }
 
-  if (/^screen|^xterm|^vt100|color|ansi|cygwin|linux/i.test(process.env.TERM)) {
-    return true;
+  if (process.env.TERM === 'dumb') {
```

Comment:
Switching the `dumb` check and the other `process.env.TERM` check does not change anything, because they are exclusive. All it does it making this diff harder to read. I suggest you switch them back to minimize this diff.